examples/rock-paper-scissors.sol

```diff
@@ -96,27 +98,41 @@ contract RockPaperScissors
         // The commitment is a hash of some salt and the hand.
         constructor (bytes32 _commitment, bytes32 _key_hash, uint _wager_amount) public payable
         {
-                // This function can only be called while at state 0.
-                require(state == uninitialized);
+                // This function can only be called while at state Uninitialized.
+                require(state == State.Uninitialized);
                 require(msg.value > _wager_amount);
 
                 // Initialize the game
+                // The amount that a player gains if they win fairly,
+                // or loses if they loose fairly
                 wager_amount = _wager_amount;
+                // The amount that player0 looses if they "cheat" by
+                // not revealing their hand after they committed.
+                // Player0 gets this back if they "play fair" by
+                // revealing their hand, whether they win or lose.
                 escrow_amount = msg.value - wager_amount;
+                // The address of player0, including to transfer to
+                // if player0 wins.
                 player0_address = msg.sender;
+                // Restructs what player0's hand can be: he has to
```

Comment:
Restricts. Or rather, commits player0 to only be able to play the preimage to this digest.

Reply:
Thanks, fixed now.